petri/src/vm/hyperv/mod.rs

```diff
@@ -387,7 +393,16 @@ impl PetriVmConfigHyperV {
             let (controller_type, controller_number) = match self.generation {
                 powershell::HyperVGeneration::One => (powershell::ControllerType::Ide, i as u32),
                 powershell::HyperVGeneration::Two => {
-                    (powershell::ControllerType::Scsi, vm.add_scsi_controller(0)?)
+                    if self.scsi_relay {
+                        // For SCSI relay, use the VTL2 relay functionality
+                        let scsi_instance_id = Guid::new_random();
+                        (
+                            powershell::ControllerType::Scsi,
+                            vm.add_scsi_controller_with_vtl2_relay(&scsi_instance_id)?,
+                        )
+                    } else {
+                        (powershell::ControllerType::Scsi, vm.add_scsi_controller(0)?)
+                    }
```

Comment:
Add a todo: comment. `PCAT` VMs can use OpenHCL for storage, but that is not implemented yet.